src/main/scala/mesosphere/marathon/api/v2/json/V2Group.scala

```diff
@@ -36,4 +42,76 @@ object V2Group {
       groups = group.groups.map(V2Group(_)),
       dependencies = group.dependencies,
       version = group.version)
+
+  implicit val v2GroupValidator: Validator[V2Group] = validator[V2Group] { group =>
+    group.id is valid
+    group.apps is valid
+    group.groups is valid
+    group.groups is noAppsAndGroupsWithSameName(group.apps)
+    group.dependencies is noCyclicDependencies(group)
```

Comment:
Good point. Should the root group still always be checked when an arbitrary `V2Group` is validated? Or should this one be extracted in a separate Validator and explicitly be called when appropriate?